src/libaktualizr/package_manager/ostreemanager.cc

```diff
@@ -93,13 +93,12 @@ data::InstallationResult OstreeManager::pull(const boost::filesystem::path &sysr
     return data::InstallationResult(data::ResultCode::Numeric::kInstallFailed, "Could not get OSTree repo");
   }
 
-  GHashTable *ref_list = nullptr;
-  if (ostree_repo_list_commit_objects_starting_with(repo.get(), refhash.c_str(), &ref_list, nullptr, &error) != 0) {
-    guint length = g_hash_table_size(ref_list);
-    g_hash_table_destroy(ref_list);  // OSTree creates the table with destroy notifiers, so no memory leaks expected
-    // should never be greater than 1, but use >= for robustness
-    if (length >= 1) {
-      LOG_DEBUG << "refhash already pulled";
+  OstreeRepoCommitState commit_state;
+  if (ostree_repo_load_commit(repo.get(), refhash.c_str(), nullptr, &commit_state, &error)) {
+    if (commit_state & OSTREE_REPO_COMMIT_STATE_PARTIAL || commit_state & OSTREE_REPO_COMMIT_STATE_FSCK_PARTIAL) {
+      LOG_INFO << "refhash " << refhash << " is partially pulled";
```

Comment:
Maybe add `... is partially pulled, re-pulling it...`?

Reply:
I wonder if we need to check whether an ostree commit/refhash is present at all and always re-pull as long as a client decides to call `OstreeManager::pull`, or make it configurable, so a client can always enforce re-pulling.

Reply:
Always re-pulling should be fine, it was my first approach to the problem. I don't see much of a reason to do the check, but I've tried minimize the behavior changes.
I also don't see a reason to make it configurable.